stetho/src/main/java/com/facebook/stetho/inspector/elements/android/ViewDescriptor.java

```diff
@@ -10,19 +10,71 @@
 package com.facebook.stetho.inspector.elements.android;
 
 import android.view.View;
+import android.view.ViewDebug;
+import com.facebook.stetho.common.Accumulator;
+import com.facebook.stetho.common.LogUtil;
 import com.facebook.stetho.common.StringUtil;
 import com.facebook.stetho.common.android.ResourcesUtil;
-import com.facebook.stetho.inspector.elements.AttributeAccumulator;
 import com.facebook.stetho.inspector.elements.AbstractChainedDescriptor;
+import com.facebook.stetho.inspector.elements.AttributeAccumulator;
+import com.facebook.stetho.inspector.elements.StyleInfo;
 
 import javax.annotation.Nullable;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
 import java.util.Map;
+import java.util.Set;
 
 final class ViewDescriptor extends AbstractChainedDescriptor<View> implements HighlightableDescriptor {
   private static final String ID_ATTRIBUTE_NAME = "id";
+  private static final Set<Method> viewGetters;
 
   private final MethodInvoker mMethodInvoker;
 
+  static {
+    Set<String> blacklistedGetters = new HashSet<>(
+      Arrays.asList(
```

Comment:
We should consider an alternate implementation for the disallowed list of method names. Instead of creating an array, then wrapping into a `List` so that we can copy it into a `HashSet` ... why don't we try this instead: Process the list of methods (`View.class.getDeclaredMethods`) without regard to whether they're allowed or disallowed. Then, before finalizing the `Set` of methods, just call `remove` for each of the disallowed methods, e.g.

```
Method[] viewMethods = ...
...
for (Method method : viewMethods) {
  if (isMethodsGetter(method) 
      && method.getReturnType().isPrimitive()) {
    getters.add(method);
  }
}

// These getters aren't allowed or something
getters.remove("getAccessibilityCursorPosition");
getters.remove("getAccessibilityWindowId");
...
getters.remove("getOverScrollMode");
getters.remove("getSystemUiVisibility");
```

This will help to eliminate a bunch of temporary allocations and CPU cycles. The one disadvantage is that the final `Set` of getters may end up with a slightly larger internal array allocation. We can still optimize that if necessary.